designer/client/page-edit.js

```diff
@@ -9,24 +9,29 @@ class PageEdit extends React.Component {
     const form = e.target
     const formData = new window.FormData(form)
     const title = formData.get('title').trim()
-    const newPath = '/'.concat(title.replace(/[^a-zA-Z ]/g, '').replace(' ', '-')).toLowerCase()
+    const newPath = '/'.concat(title.replace(/[^a-zA-Z ]/g, '').replace(/\s+/g, '-')).toLowerCase()
     const section = formData.get('section').trim()
     const pageType = formData.get('page-type').trim()
     const { data, page } = this.props
 
     const copy = clone(data)
-    const pathChanged = newPath !== page.path
+    const titleChanged = title !== page.title
     const pageIndex = data.pages.indexOf(page)
     const copyPage = copy.pages[pageIndex]
 
-    if (pathChanged) {
+    if (titleChanged) {
       // `path` has changed - validate it is unique
-      if (data.findPage(newPath)) {
-        form.elements.path.setCustomValidity(`Path '${newPath}' already exists`)
+      if (data.findPage(title)) {
+        form.elements.path.setCustomValidity(`Title '${title}' already exists`)
         form.reportValidity()
         return
       }
-
+      // we want to also update the `next:` references for this page with the new path to preserve the link
+      copy.pages.filter(page => page.next.map(next => {
```

Comment:
this is doing a filter on an array but does not do a match, and then you don't do anything with the result of filter. Did you mean to do something like forEach? 

Your map call is also returning an array of [null, null, null...] probably since there's no return call.